### DISCUSSION TITLE
Prisma plugin - t.relation resolve to a single record?

### DISCUSSION THREAD

**Discussion by PatricioNG** [hayes/pothos#1228] [Q&A] [answered]
Wondering if there's a way to do a resolver for a relation field? Use case: a table of metadata, I have a field called `metadata` where you can resolve it all, but I wanted to expose `metafield` with an arg to query just a single metadata record and return it instead of an array

Currently I am doing this which returns the record as the first index of an array:

```
... //Rest of the prismaObject type definition

metafield: t.relation('Metadata', {
			args: {
				key: t.arg.string({ required: true }),
			},
			query: ({ key }) => ({
				where: {
					key,
				},
			}),
		}),
```

Was hoping for something like:

```
metafield: t.relation('Metadata', {
			args: {
				key: t.arg.string({ required: true }),
			},
			query: ({ key }) => ({
				where: {
					key,
				},
			}),
			nullable: true,
			resolve(_query, {Metadata}, args, context, info) {
				return Metadata[0] ?? null
			},
		}),
```

**Comment by hayes** [comment, marked as answer]
You can use t.field with `select` for this.  Something like this is probably what you are looking for:

```typescript
const MetadataRef = builder.prismaObject('Metadata', { /* ... */ })

// ...
(t) => ({
  metafield: t.field({
    type: MetadataRef,
    nullable: true,
    args: {
      key: t.arg.string({ required: true }),
    },
    select: ({ key }, ctx, nestedSelection) => ({
      Metadata: nestedSelection({
        where: {
          key,
        }
      })
    }),
    resolve({ Metadata }, args, context, info) {
      return Metadata[0] ?? null;
    },
  }),
});
```

**Reply by PatricioNG** [reply]
Ah incredible, thanks @hayes! Appreciate how responsive you are for this, it's super helpful! 🙏 🙌

**Reply by hayes** [reply]
No problem.  If you have time, I'm looking for people willing to go through the 4.0 migration before the final release to get some feedback and make sure the important stuff is all covered: https://v4.pothos-graphql.dev/docs/migrations/v4
